packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -214,8 +214,21 @@ type ApplyFromHistoryAction = {
   query: BaseQuery;
 };
 
-export const applyFromHistory = (query: BaseQuery): ApplyFromHistoryAction => {
-  return { type: QueryBarActions.ApplyFromHistory, query };
+export const applyFromHistory = (
+  query: BaseQuery & { update?: Document }
+): QueryBarThunkAction<Promise<void>, ApplyFromHistoryAction> => {
+  return async (dispatch, getState, { globalAppRegistry }) => {
+    dispatch({
+      type: QueryBarActions.ApplyFromHistory,
+      query,
+    } as ApplyFromHistoryAction);
+
+    if (query.update) {
+      globalAppRegistry?.emit('favorites:open-bulk-update-favorite', query);
```

Comment:
This should be `localAppRegistry`, not global one. Emitting this on global means that every tab currently in runtime will receive the event and will open the modal, not only the one from which the action was actually triggered